tests/v1/entrypoints/llm/test_struct_output_generate.py

```diff
@@ -501,6 +505,40 @@ def test_structured_output(
             pytest.fail("Invalid function call format: "
                         f"{generated_text!r}\nError: {str(e)}")
 
+    #
+    # Test 12: Generate structured output with reasoning step
+    #
+    if reasoning_parser is not None:
+        reasoning_prompt = "Solve the following math problem step-by-step, then provide the final answer as JSON object with a single key 'result'. Problem: What is 5 * 8 + 2?"  # noqa: E501
+        reasoning_schema = {
+            "type": "object",
+            "properties": {
+                "result": {
+                    "type": "integer"
+                }
+            },
+            "required": ["result"]
+        }
+
+        sampling_params = SamplingParams(
+            temperature=0.1,
+            max_tokens=200,
```

Comment:
Since we get failures sometimes due to cutting off the output before the model completes a valid JSON object.

```suggestion
            max_tokens=4096,
```